### PR TITLE
Pass auth credentials using env vars

### PR DESCRIPTION
**Pull request by jeremy**
Pass Sauce credentials to child process using env vars so they don't show up on the command line.

Adds some docs on how to authenticate and configure.

### CODE REVIEW

File: lib/blade/sauce_labs_plugin.rb [hunk 1]
```diff
@@ -48,11 +48,11 @@ def config
   end
 
   def username
-    config.username || ENV["SAUCE_USERNAME"]
+    ENV["SAUCE_USERNAME"] || config.username
   end
 
   def access_key
-    config.access_key || ENV["SAUCE_ACCESS_KEY"]
+    ENV["SAUCE_ACCESS_KEY"] || config.access_key
```
Comment (jeremy) [inline lib/blade/sauce_labs_plugin.rb:55]: Flip so environment-provided credentials always override whatever's in `.blade.yml` config

File: lib/blade/sauce_labs_plugin/tunnel.rb [hunk 2]
```diff
@@ -36,6 +36,7 @@ def create_child_process
       ChildProcess.build(tunnel_command, *tunnel_args).tap do |process|
         process.leader = true
         process.io.inherit! if debug?
+        process.environment.merge! tunnel_env
```
Comment (jeremy) [inline lib/blade/sauce_labs_plugin/tunnel.rb:39]: Inject auth credentials into child process' environment. It inherits existing env vars, but this ensures that username/access_key in .blade.yml will be passed as env vars also.

File: README.md [hunk 3]
```diff
@@ -1 +1,51 @@
 [Blade](https://github.com/javan/blade) plugin for [Sauce Labs](https://saucelabs.com/)
+
+## Configuration
+
+### Authenticate with Sauce Labs
+
+Set the `SAUCE_USERNAME` and `SAUCE_API_KEY` environment variables to authenticate with Sauce Labs.
+
+All CI tools provide a way to set environment variables for a test run. For non-CI test runs, set the environment variables in your shell or in your test runner script.
```
Comment (jeremy) [inline README.md:9]: Can pass credentials in `.blade.yml` also, but figure we should focus on the preferred way to provide them (externally, not embedded in code)

File: README.md [hunk 4]
```diff
@@ -1 +1,51 @@
...
+      # Specific iOS Mobile Safari versions:
+      iPhone:
+        version: [9.2, 8.4]
+
+      # Mobile-specific browser:
+      Motorola Droid 4 Emulator:
+        version: [5.1, 4.4]
+```
+
+See Sauce Labs' [Platform Configurator](https://wiki.saucelabs.com/display/DOCS/Platform+Configurator) for an exhaustive list of supported devices, operating systems, and browsers.
```
Comment (jeremy) [inline README.md:51]: Quick stab at explaining the platform matching heuristics.

**Comment by javan** [comment]
❤️❤️

**Pull request merged**